api/v1beta2/condition_types.go

```diff
@@ -19,15 +19,24 @@ package v1beta2
 const SourceFinalizer = "finalizers.fluxcd.io"
 
 const (
+	// ArtifactInStorageCondition indicates the availability of Artifact of the
+	// Source in the storage.
+	// If True, the Artifact is stored successfully. If False, the Artifact
+	// couldn't be stored.
+	// This Condition is only present on the resource if the Artifact is
+	// successfully stored.
+	ArtifactInStorageCondition string = "ArtifactInStorage"
+
 	// ArtifactOutdatedCondition indicates the current Artifact of the Source
 	// is outdated.
 	// This is a "negative polarity" or "abnormal-true" type, and is only
 	// present on the resource if it is True.
 	ArtifactOutdatedCondition string = "ArtifactOutdated"
 
-	// SourceVerifiedCondition indicates the integrity of the Source has been
-	// verified. If True, the integrity check succeeded. If False, it failed.
-	// The Condition is only present on the resource if the integrity has been
+	// SourceVerifiedCondition indicates the integrity verification of the
+	// Source.
+	// If True, the integrity check succeeded. If False, it failed.
+	// This Condition is only present on the resource if the integrity has been
```

Comment:
Same concern as above.

Reply:
This statement is not contradictory. If the declaration on the resource is to check the signature of a commit (or check the integrity in another way), the Condition will be present in either a True or False state.

Reply:
Yes, but the wording can be misleading. I think it'd be better if it said something like `the Condition is present only when we attempt to verify it's integrity` because the current wording implies that it's present only if it's been verified.